internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -25,13 +25,18 @@ type LegacySchemaInfo struct {
 	RootDiskSize               *float64
 }
 
-func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, timeout timeouts.Value, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo) *TFModel {
+type ExtraAPIInfo struct {
+	ContainerIDs map[string]string
+}
+
+func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, timeout timeouts.Value, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo, apiInfo *ExtraAPIInfo) *TFModel {
```

Comment:
I feel like we're overusing pointer params, e.g. why apiInfo is a pointer, it it optional?.

Also apiInfo could have zero value, not being a pointer, and have ContainerIDs to null is optional

Reply:
Since it is passed in multiple nested functions, I thought it was better to use a pointer.
Changed it now.

An alternative, is to store in the `ctx`, WDYT?

Reply:
i wouldn't use ctx.  i think it's ok now